File: pages/api/tweet-writer.js
import { generateTweets } from '../../services/tweetWriter';

export default async function handler(req, res) {
  if (req.method === 'POST') {
    try {
      const text = req.body.text;
      const tweets = await generateTweets(text);
      res.status(200).json(tweets);
    } catch (error) {
      res.status(500).json({ error: 'An error occurred while processing the request.' });
    }
  } else {
    res.status(405).json({ error: 'Method not allowed.' });
  }
}

File: services/TweetWriter.py
from flask import Flask, request, jsonify
import tweet_writer

app = Flask(__name__)

@app.route('/api/tweet-writer', methods=['POST'])
def generate_tweets():
    text = request.json.get('text', '')
    tweets = tweet_writer.generate_tweets(text)
    return jsonify(tweets)

if __name__ == '__main__':
    app.run()
